cpp/src/copying/concatenate.cu

```diff
@@ -360,6 +369,16 @@ std::unique_ptr<table> concatenate(std::vector<table_view> const& tables_to_conc
                            }),
                "Mismatch in table columns to concatenate.");
 
+  if (is_fixed_point(tables_to_concat.front().column(0).type())) {
+    CUDF_EXPECTS(std::adjacent_find(tables_to_concat.cbegin(),
+                                    tables_to_concat.cend(),
+                                    [](auto const& l, auto const& r) {
+                                      return l.column(0).type().scale() !=
+                                             r.column(0).type().scale();
```

Comment:
only first column scales are compared. What about rest of the columns?

Reply:
tables should always have columns with the same scale, so only need to check 1

Reply:
is this enforced in all tables or just input `table_view` to `cudf::concatenate`? how is this enforced?

Reply:
Good point, let me check.

Reply:
Let me set up a follow up PR to address this. I locally added the following code:
```cpp
table_view_base<ColumnView>::table_view_base(std::vector<ColumnView> const& cols) : _columns{cols}
{
  if (num_columns() > 0) {
    CUDF_EXPECTS(
      std::all_of(_columns.cbegin(),
                  _columns.cend(),
                  [size = _columns.front().size()](auto const& col) { return col.size() == size; }),
      "Column size mismatch.");

    if (is_fixed_point(_columns.front().type())) {
      CUDF_EXPECTS(std::all_of(_columns.cbegin(),
                               _columns.cend(),
                               [scale = _columns.front().type().scale()](auto const& c) {
                                 return scale == c.type().scale();
                               }),
                   "fixed_point table_view columns must have the same scale.");
    }
    _num_rows = _columns.front().size();
  } else {
    _num_rows = 0;
  }
}
```
and a unit test in `table_tests.cpp`
```cpp
struct FixedPointTest : public cudf::test::BaseFixture {
};

TEST_F(FixedPointTest, FixedPointScaleMismatch)
{
  using namespace numeric;
  using fp_wrapper = cudf::test::fixed_point_column_wrapper<int32_t>;

  auto begin     = thrust::make_counting_iterator(0);
  auto const vec = std::vector<int32_t>(begin, begin + 600);

  auto const a = fp_wrapper(vec.begin(), /***/ vec.begin() + 200, scale_type{-1});
  auto const b = fp_wrapper(vec.begin() + 200, vec.begin() + 400, scale_type{-2});
  auto const c = fp_wrapper(vec.begin() + 400, vec.end(), /*****/ scale_type{-3});

  auto const fixed_point_columns = std::vector<cudf::column_view>{a, b, c};
  EXPECT_THROW(cudf::table_view{fixed_point_columns}, cudf::logic_error);
}
```
However, this breaks binary ops because `bitmask_and` takes a `table_view` when it really shouldn't. It should take a `std::vector` of `cudf::column`s. 

It will be cleaner to sort this out and follow up PR.

Reply:
I started the PR

Reply:
PR / Issue and my understanding was wrong. I will add a more comprehensive check here :+1:

Reply:
And by more comprehensive - it actually just means removing the check for table_view, as the column types don't need to be the same.